diff --git a/src/cmd/flux-exec.c b/src/cmd/flux-exec.c
index b7fadafb5011..2f2cf16566d4 100644
--- a/src/cmd/flux-exec.c
+++ b/src/cmd/flux-exec.c
@@ -208,9 +208,11 @@ void output_cb (flux_subprocess_t *p, const char *stream)
         log_err_exit ("flux_subprocess_getline");
 
     if (lenp) {
+        int rc;
         if (optparse_getopt (opts, "label-io", NULL) > 0)
             fprintf (fstream, "%d: ", flux_subprocess_rank (p));
-        fwrite (ptr, lenp, 1, fstream);
+        rc = fwrite (ptr, lenp, 1, fstream);
+        (void) rc;
     }
 }
 
diff --git a/src/common/libflux/message.c b/src/common/libflux/message.c
index 7191d3996e40..765dda11c3f0 100644
--- a/src/common/libflux/message.c
+++ b/src/common/libflux/message.c
@@ -1241,12 +1241,10 @@ static void flags2str (int flags, char *buf, int buflen)
 
 static void userid2str (uint32_t userid, char *buf, int buflen)
 {
-    int n;
     if (userid == FLUX_USERID_UNKNOWN)
-        n = snprintf (buf, buflen, "unknown");
+        snprintf (buf, buflen, "unknown");
     else
-        n = snprintf (buf, buflen, "%u", userid);
-    assert (n < buflen);
+        snprintf (buf, buflen, "%u", userid);
 }
 
 static int roletostr (uint32_t role, const char *sep, char *buf, int buflen)
@@ -1286,14 +1284,12 @@ static void rolemask2str (uint32_t rolemask, char *buf, int buflen)
 
 static void nodeid2str (uint32_t nodeid, char *buf, int buflen)
 {
-    int n;
     if (nodeid == FLUX_NODEID_ANY)
-        n = snprintf (buf, buflen, "any");
+        snprintf (buf, buflen, "any");
     else if (nodeid == FLUX_NODEID_UPSTREAM)
-        n = snprintf (buf, buflen, "upstream");
+        snprintf (buf, buflen, "upstream");
     else
-        n = snprintf (buf, buflen, "%u", nodeid);
-    assert (n < buflen);
+        snprintf (buf, buflen, "%u", nodeid);
 }
 
 void flux_msg_fprint_ts (FILE *f, const flux_msg_t *msg, double timestamp)
diff --git a/src/common/libsubprocess/subprocess.c b/src/common/libsubprocess/subprocess.c
index 1c3a313a2778..ec93c40946dd 100644
--- a/src/common/libsubprocess/subprocess.c
+++ b/src/common/libsubprocess/subprocess.c
@@ -263,8 +263,10 @@ void flux_standard_output (flux_subprocess_t *p, const char *stream)
         }
     }
 
-    if (lenp)
-        fwrite (ptr, lenp, 1, fstream);
+    if (lenp) {
+        int rc = fwrite (ptr, lenp, 1, fstream);
+        (void) rc;
+    }
 }
 
 /*
diff --git a/src/modules/kvs/cache.c b/src/modules/kvs/cache.c
index 2b6d05c817da..07c1b0c4bd7c 100644
--- a/src/modules/kvs/cache.c
+++ b/src/modules/kvs/cache.c
@@ -365,10 +365,9 @@ struct cache_entry *cache_lookup (struct cache *cache, const char *ref)
 
 int cache_insert (struct cache *cache, struct cache_entry *entry)
 {
-    int rc;
 
     if (cache && entry) {
-        rc = zhashx_insert (cache->zhx, entry->blobref, entry);
+        zhashx_insert (cache->zhx, entry->blobref, entry);
         list_add (&cache->entries_list, &entry->entries_node);
         entry->notdirty_list = &cache->notdirty_list;
         entry->valid_list = &cache->valid_list;
@@ -378,7 +377,6 @@ int cache_insert (struct cache *cache, struct cache_entry *entry)
         if (entry->waitlist_valid
             && wait_queue_msgs_count (entry->waitlist_valid) > 0)
             list_add (entry->valid_list, &entry->valid_node);
-        assert (rc == 0);
     }
     return 0;
 }
diff --git a/src/shell/output.c b/src/shell/output.c
index dc3b4260326b..4fa9490e3cde 100644
--- a/src/shell/output.c
+++ b/src/shell/output.c
@@ -177,8 +177,10 @@ static int shell_output_term (struct shell_output *out)
                 f = stderr;
             }
             if ((output_type == FLUX_OUTPUT_TYPE_TERM) && len > 0) {
+                int rc;
                 fprintf (f, "%s: ", rank);
-                fwrite (data, len, 1, f);
+                rc = fwrite (data, len, 1, f);
+                (void) rc;
             }
             free (data);
         }
diff --git a/t/rexec/rexec_count_stdout.c b/t/rexec/rexec_count_stdout.c
index 49a942586e27..cce9341c808f 100644
--- a/t/rexec/rexec_count_stdout.c
+++ b/t/rexec/rexec_count_stdout.c
@@ -70,8 +70,10 @@ void output_cb (flux_subprocess_t *p, const char *stream)
         }
     }
 
-    if (lenp)
-        fwrite (ptr, lenp, 1, fstream);
+    if (lenp) {
+        int rc = fwrite (ptr, lenp, 1, fstream);
+        (void) rc;
+    }
 
     if (!strcasecmp (stream, "stdout"))
         stdout_count++;
diff --git a/t/rexec/rexec_getline.c b/t/rexec/rexec_getline.c
index 9171b02242c6..65c30fe1762f 100644
--- a/t/rexec/rexec_getline.c
+++ b/t/rexec/rexec_getline.c
@@ -74,8 +74,10 @@ void output_cb (flux_subprocess_t *p, const char *stream)
 
     if (!(ptr = flux_subprocess_getline (p, stream, &lenp)))
         log_err_exit ("flux_subprocess_getline");
-    if (lenp)
-        fwrite (ptr, lenp, 1, fstream);
+    if (lenp) {
+        int rc = fwrite (ptr, lenp, 1, fstream);
+        (void) (rc);
+    }
     else
         fprintf (fstream, "EOF\n");
 }
