From 1f0ff493aad771016a763f6a5cbb97ae489161c2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jan=20Jan=C3=9Fen?= <janssen@mpie.de>
Date: Fri, 3 Oct 2025 16:40:28 +0200
Subject: [PATCH] zero message queue patch

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index d7d513f..cf5e333 100644
--- a/configure.ac
+++ b/configure.ac
@@ -333,7 +333,7 @@ AM_CONDITIONAL([ENABLE_PYLINT], [test "x$PYLINT" = "xpylint"])
 AX_PROG_LUA([5.1],[5.5])
 AX_LUA_HEADERS
 AX_LUA_LIBS
-X_AC_ZEROMQ
+PKG_CHECK_MODULES([ZMQ], [libzmq >= 4.0.4])
 X_AC_JANSSON
 PKG_CHECK_MODULES([LIBSYSTEMD], [libsystemd >= 0.23],
                                 [have_libsystemd=yes], [have_libsystemd=no])
