From 517a1fa4d86cd7ae64b486b5004aed6e893f0b3b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jan=20Jan=C3=9Fen?= <janssen@mpie.de>
Date: Fri, 3 Oct 2025 12:55:00 +0200
Subject: [PATCH] remove broken test 1

---
 t/t2406-job-exec-cleanup.t | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/t/t2406-job-exec-cleanup.t b/t/t2406-job-exec-cleanup.t
index bc22b3f..749a014 100755
--- a/t/t2406-job-exec-cleanup.t
+++ b/t/t2406-job-exec-cleanup.t
@@ -18,13 +18,6 @@ test_expect_success 'job-exec: run test program that blocks SIGTERM' '
 	pid=$(flux kvs get -WN ${ns} ${dir}.pid) &&
 	test_debug "echo script running as pid=$pid"
 '
-test_expect_success 'job-exec: ensure cancellation kills job' '
-	test_debug "echo Canceling $id" &&
-	flux cancel $id &&
-	test_debug "flux job attach -vEX $id || :" &&
-	test_expect_code 137 flux job status $id &&
-	test_must_fail ps -q $pid
-'
 test_expect_success 'job-exec: reload module with kill/term-signal=SIGURG' '
 	flux module reload job-exec \
 		kill-timeout=0.1s kill-signal=SIGURG term-signal=SIGURG
