From 48f38c75c674343523305eb9c34b6b99219c26fe Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jan=20Jan=C3=9Fen?= <janssen@mpie.de>
Date: Sat, 4 Oct 2025 17:26:48 +0200
Subject: [PATCH] Remove python/t0002-wrapper.py

---
 t/Makefile.am | 1 -
 1 file changed, 1 deletion(-)

diff --git a/t/Makefile.am b/t/Makefile.am
index a24d2dc..dd501f5 100644
--- a/t/Makefile.am
+++ b/t/Makefile.am
@@ -284,7 +284,6 @@ TESTSCRIPTS = \
 	lua/t1001-timeouts.t \
 	lua/t1004-statwatcher.t \
 	python/t0001-handle.py \
-	python/t0002-wrapper.py \
 	python/t0004-event.py \
 	python/t0005-kvs.py \
 	python/t0006-request.py \
