From d281a6664ed24f428880531943acafc468fe6a82 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jan=20Jan=C3=9Fen?= <janssen@mpie.de>
Date: Fri, 3 Oct 2025 10:38:04 +0200
Subject: [PATCH] remove t2273-job-alloc-bypass

---
 t/Makefile.am | 1 -
 1 file changed, 1 deletion(-)

diff --git a/t/Makefile.am b/t/Makefile.am
index 522b463..7d05d9e 100644
--- a/t/Makefile.am
+++ b/t/Makefile.am
@@ -158,7 +158,6 @@ TESTSCRIPTS = \
 	t2270-job-dependencies.t \
 	t2271-job-dependency-after.t \
 	t2272-job-begin-time.t \
-	t2273-job-alloc-bypass.t \
 	t2274-manager-perilog.t \
 	t2275-job-duration-validator.t \
 	t2276-job-requires.t \
