From 4121f254806094736b544be0b2297d7499b63bec Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jan=20Jan=C3=9Fen?= <janssen@mpie.de>
Date: Fri, 3 Oct 2025 11:57:08 +0200
Subject: [PATCH] Disable t2271-job-dependency-after.t

---
 t/Makefile.am | 1 -
 1 file changed, 1 deletion(-)

diff --git a/t/Makefile.am b/t/Makefile.am
index 449299d..0fe6eda 100644
--- a/t/Makefile.am
+++ b/t/Makefile.am
@@ -156,7 +156,6 @@ TESTSCRIPTS = \
 	t2261-job-list-update.t \
 	t2262-job-list-stats.t \
 	t2270-job-dependencies.t \
-	t2271-job-dependency-after.t \
 	t2274-manager-perilog.t \
 	t2275-job-duration-validator.t \
 	t2276-job-requires.t \
