From 5e076fa49b8c4f0a3021783838af8104ec70952e Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Thu, 4 Jun 2026 08:07:24 +0200
Subject: [PATCH] Remove t0100-modprobe.t from Makefile

---
 t/Makefile.am | 1 -
 1 file changed, 1 deletion(-)

diff --git a/t/Makefile.am b/t/Makefile.am
index 352efcda1ac0..40e8ee004544 100644
--- a/t/Makefile.am
+++ b/t/Makefile.am
@@ -108,7 +108,6 @@ TESTSCRIPTS = \
 	t0042-rhwloc-gpu-dedup.t \
 	t0090-content-enospc.t \
 	t0099-admin-system-scripts.t \
-	t0100-modprobe.t \
 	t0800-completion.t \
 	t1000-kvs.t \
 	t1001-kvs-internals.t \
