From cb40faec5281fdc4e2d6ba9c2c584b92b056a75a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jan=20Jan=C3=9Fen?= <janssen@mpie.de>
Date: Fri, 3 Oct 2025 15:23:49 +0200
Subject: [PATCH] Remove perilog

---
 t/Makefile.am | 1 -
 1 file changed, 1 deletion(-)

diff --git a/t/Makefile.am b/t/Makefile.am
index adcecbc..59f7833 100644
--- a/t/Makefile.am
+++ b/t/Makefile.am
@@ -161,7 +161,6 @@ TESTSCRIPTS = \
 	t2271-job-dependency-after.t \
 	t2272-job-begin-time.t \
 	t2273-job-alloc-bypass.t \
-	t2274-manager-perilog.t \
 	t2274-manager-perilog-per-rank.t \
 	t2275-job-duration-validator.t \
 	t2276-job-requires.t \
