From 22009d5f1c8e0cea0bb832225002831882f40e74 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jan=20Jan=C3=9Fen?= <janssen@mpie.de>
Date: Fri, 3 Oct 2025 12:55:10 +0200
Subject: [PATCH] remove broken test 2

---
 t/t2802-uri-cmd.t | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/t/t2802-uri-cmd.t b/t/t2802-uri-cmd.t
index b39c784..2045f60 100755
--- a/t/t2802-uri-cmd.t
+++ b/t/t2802-uri-cmd.t
@@ -165,12 +165,6 @@ test_expect_success 'setup fake csm_allocation_query for mock lsf testing' '
 	export CSM_ALLOCATION_QUERY=$(pwd)/csm_allocation_query &&
 	export FLUX_SSH=${SHARNESS_TEST_SRCDIR}/scripts/tssh
 '
-test_expect_success 'flux-uri mock testing of lsf resolver works' '
-	result=$(SHELL=/bin/sh flux uri --local lsf:12345) &&
-	FLUX_URI_1=$(flux exec -r 1 flux getattr local-uri) &&
-	test_debug "echo checking if $result is $FLUX_URI or $FLUX_URI_1" &&
-	test "$result" = "$FLUX_URI" -o "$result" = "$FLUX_URI_1"
-'
 test_expect_success 'cleanup jobs' '
 	flux cancel --all &&
 	flux queue drain
