From d2848956c76ba59b2bdc48ae00acd740ba08907a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jan=20Jan=C3=9Fen?= <janssen@mpie.de>
Date: Fri, 3 Oct 2025 16:22:06 +0200
Subject: [PATCH] python patch

---
 t/Makefile.am | 1 -
 1 file changed, 1 deletion(-)

diff --git a/t/Makefile.am b/t/Makefile.am
index 87d1509..6797bec 100644
--- a/t/Makefile.am
+++ b/t/Makefile.am
@@ -262,7 +262,6 @@ TESTSCRIPTS = \
 	lua/t1001-timeouts.t \
 	lua/t1004-statwatcher.t \
 	python/t0001-handle.py \
-	python/t0002-wrapper.py \
 	python/t0003-barrier.py \
 	python/t0004-event.py \
 	python/t0005-kvs.py \
