From 56ec50772142f0ad811bedfc9bb9b9b4dbe75ceb Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jan=20Jan=C3=9Fen?= <janssen@mpie.de>
Date: Fri, 3 Oct 2025 09:37:46 +0200
Subject: [PATCH] Disable t0013-config-file.t

---
 t/Makefile.am | 1 -
 1 file changed, 1 deletion(-)

diff --git a/t/Makefile.am b/t/Makefile.am
index 87d1509..522b463 100644
--- a/t/Makefile.am
+++ b/t/Makefile.am
@@ -84,7 +84,6 @@ TESTSCRIPTS = \
 	t0025-broker-state-machine.t \
 	t0027-broker-groups.t \
  t0040-flux-sproc.t \
-	t0013-config-file.t \
 	t0014-runlevel.t \
 	t0015-cron.t \
 	t0016-cron-faketime.t \
